client/lib/analytics/ad-tracking/record-order.js

```diff
@@ -540,6 +543,51 @@ function recordOrderInWPcomGA4( cart, orderId, wpcomJetpackCartInfo ) {
 	debug( 'recordOrderInWPcomGA4: Record WPcom Purchase in GA4' );
 }
 
+/**
+ * Sends a purchase event to Google Tag Manager for eligible Woo Express upgrades.
+ *
+ * @param {import('@automattic/shopping-cart').ResponseCart} cart
+ * @param {string} orderId
+ * @param {import('@automattic/data-stores/src').SiteDetails} selectedSiteData
+ * @returns {void}
+ */
+function recordOrderInWooGTM( cart, orderId, selectedSiteData ) {
+	if (
+		! mayWeTrackByTracker( 'googleTagManager' ) ||
+		! isWooExpressUpgrade( cart, selectedSiteData )
+	) {
+		return;
+	}
+	loadWooGTMContainer()
+		.then( () => initWooGTMContainer() )
+		.then( () => {
+			debug(
+				`recordOrderInWooGTM: Initialized GTM container ${ TRACKING_IDS.wooGoogleTagManagerId }`
+			);
+			const purchaseEventMeta = {
+				event: 'purchase',
+				ecommerce: {
+					coupon: cart.coupon?.toString() ?? '',
+					transaction_id: orderId,
+					currency: 'USD',
+					items: cart.products.map( ( { product_id, product_name, cost, volume } ) => ( {
+						id: product_id.toString(),
+						name: product_name.toString(),
+						quantity: parseInt( volume ),
+						price: costToUSD( cost, cart.currency ) ?? 0,
+						billing_term: cart.billing?.interval_unit,
+					} ) ),
+					value: cart.total_cost_integer / 100,
```

Comment:
I don't fully understand what these stats are used for but not every currency can use this math safely. In USD, `cart.total_cost_integer / 100` will return dollars because USD has two decimal places, but JPY has no decimal places, which means this would produce a strange result that doesn't have much to do with the actual cost. Is that safe here?

Reply:
I think this implementation is based on the existing code that has served us for e-commerce analytics for some time:
https://github.com/Automattic/wp-calypso/blob/add/hotjar-tracking-woo-express-plans-page/client/lib/analytics/utils/split-wpcom-jetpack-cart-info.ts#L23

However, when we track down the usage of this data - it appears we actually never use the raw value - and as you mentioned, probably this value in general is flawed.

The `value` e.g. in purchase event is converted to USD. I think we also need to use it similarly to get a valid output:
`costToUSD( cart.total_cost_integer / 100, cart.currency )`

Reply:
Cool. Just wanted to mention it. Probably `costToUSD( cost, cart.currency )`, but yes.